irr/src/SkinnedMesh.cpp

```diff
@@ -146,21 +76,15 @@ void SkinnedMesh::animateMesh(f32 frame)
 		// The joints can be animated here with no input from their
 		// parents, but for setAnimationMode extra checks are needed
 		// to their parents
-
-		core::vector3df position = joint->Animatedposition;
-		core::vector3df scale = joint->Animatedscale;
-		core::quaternion rotation = joint->Animatedrotation;
-
-		// Note: This interpolates as necessary.
-		getFrameData(frame, joint,
-				position, joint->positionHint,
-				scale, joint->scaleHint,
-				rotation, joint->rotationHint);
-
-		// No blending needed
-		joint->Animatedposition = position;
-		joint->Animatedscale = scale;
-		joint->Animatedrotation = rotation;
+		// FIXME interpolation does not seem to be working properly,
```

Comment:
The issue in question. Maybe what's going on is more obvious to a second pair of eyes.

In particular I don't see why the old code (see `getFrameData`) would have worked when this doesn't.

Reply:
Nevermind, Irrlicht was just using inconsistent interpolation conventions: Vector interpolation does $a(1 - t) + bt$, whereas quaternion slerp does $at + (1 - t)b$. Yum.